lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,267 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+                      Value indexInIndices = args[0];
+                      Value offsetsI = args[1];
+                      Value initTensorElem = args[2];
+
+                      Value indexI =
+                          rewriter.create<linalg::IndexOp>(loc, /*value=*/0);
+                      Value indexIToInt = castIndexToInt64(rewriter, loc, indexI);
+                      Value one = getConstant(
+                          rewriter, loc, 1,
+                          mlir::IntegerType::get(getContext(), 64,
+                                                IntegerType::Signless));
+                      Value offsetIndexPlusOneInt =
+                          rewriter.create<arith::AddIOp>(loc, indexIToInt, one);
+
+                      Value offsetIndexPlusOne =
+                          castIntToIndex(rewriter, loc, offsetIndexPlusOneInt);
+                      Value checkLast = rewriter.create<arith::CmpIOp>(
+                          loc, arith::CmpIPredicate::eq,
+                          castIndexToInt64(rewriter, loc, offsetsLength),
+                          offsetIndexPlusOneInt);
+                      Value nextOffset = rewriter.create<arith::SelectOp>(
+                          loc, checkLast,
+                          castIndexToInt64(rewriter, loc, indicesLength),
+                          rewriter.create<tensor::ExtractOp>(loc, offsets,
+                                                            offsetIndexPlusOne));
+
+                      Value indicesIndex = castIndexToInt64(
+                          rewriter, loc,
+                          rewriter.create<linalg::IndexOp>(loc, /*value=*/1));
+
+                      Value offsetLessThanIndicesIndex =
+                          rewriter.create<arith::CmpIOp>(
+                              loc, arith::CmpIPredicate::slt, offsetsI,
+                              indicesIndex);
+                      Value offsetEqualToIndicesIndex =
+                          rewriter.create<arith::CmpIOp>(loc,
+                                                        arith::CmpIPredicate::eq,
+                                                        offsetsI, indicesIndex);
+                      Value offsetLessThanOrEqualToIndicesIndex =
+                          rewriter.create<arith::OrIOp>(
+                              loc, offsetLessThanIndicesIndex,
+                              offsetEqualToIndicesIndex);
+
+                      Value indicesIndexLessThanNextOffset =
+                          rewriter.create<arith::CmpIOp>(
+                              loc, arith::CmpIPredicate::slt, indicesIndex,
+                              nextOffset);
+
+                      Value indicesIndexWithinBounds =
+                          rewriter.create<arith::AndIOp>(
+                              loc, offsetLessThanOrEqualToIndicesIndex,
+                              indicesIndexLessThanNextOffset);
+
+                      SmallVector<Value> index_into_weight;
```

Comment:
nit: convention is to use camel-case. There are other in these changes that also don't follow this convention. Please change all of them